src/content_sao.cpp

```diff
@@ -1273,6 +1273,42 @@ std::string PlayerSAO::getWieldList() const
 	return "main";
 }
 
+ItemStack PlayerSAO::getWieldedItem() const
+{
+	const Inventory *inv = getInventory();
+	ItemStack ret;
+	if (inv) {
+		const InventoryList *mlist = inv->getList(getWieldList());
+		if (mlist && (getWieldIndex() < (s32)mlist->getSize()))
+			ret = mlist->getItem(getWieldIndex());
+		if (ret.name == "") {
+			const InventoryList *hlist = inv->getList("hand");
+			if (hlist)
+				ret = hlist->getItem(0);
+		}
+	}
+	return ret;
```

Comment:
No. ret is only set to hlist->getItem(0) if the name of the current selected item of the hotbar is "".
See https://github.com/minetest/minetest/pull/2827#discussion-diff-46604579R1283